### DISCUSSION TITLE
Question regarding custom Nc Neuron Models

### DISCUSSION THREAD

**Discussion by Andy-Urbano** [lava-nc/lava#680] [General]
I am currently trying to create a custom neuron model to run on the Loihi2. However, I am having a lot of difficulty with the implementation of micro-coded neurons, as there are no tutorials. 
Specifically:
- Is the Instruction set for the Loihi2 publicly available?
- Is it explained anywhere how to connect the NcInPorts of a process model to the "InPorts" of a Nodes instance created by the Net module? All of the examples I've seen only have one source of input, what if I want to have a neuron integrate inputs from two different sources differently?
- Is there any tutorial that explains all the different components of the Net Module and how to use them? If not, where could I get this info?

Thank you very much for any help you may be able to provide. :)

**Comment by weidel-p** [comment]
Hi @Andy-Urbano,

thanks for your message. 

- The instruction set for microcoded neurons is not publicly available. If you want to write your own microcoded neuron model, please write an email to [nrc_support@intel-research.net](mailto:nrc_support@intel-research.net.) requesting access to the documentation of the loihi2 instruction set. Please add a few lines stating who you are, your INRC membership status and your project.
- In the currently released version of lava-loihi, it is not possible to integrate inputs from multiple sources. This feature is currently under development and you can expect to see this feature in the next release. If you need this feature urgently, we can give you access to the development version.
- Documentation about these topics is rather sparse. Please take a look into the tutorials in the lava-loihi repository and the documentation you find on the nfs on vlab. If that does not answer your questions, I recommend looking at the tests and existing Processes/Models. Of course you can always reach out here or via the email above, we are always happy to assist.
